Sorbet/main.rb

```diff
@@ -0,0 +1,355 @@
+  else
+    0
+  end
+end
+
+## failure
+sig { params(x: T.untyped).returns(Integer) }
+def alias_failure_f(x)
+  y = x.is_a?(String)
+  if y
+    x.is_nan # Expected error: No method 'is_nan' on String
+  else
+    0
+  end
+end
+
+## failure
+sig { params(x: T.any(String, Integer)).returns(Integer) }
+def alias_failure_g(x)
+  y = x.is_a?(String)
+  if y
+    x.length
+  else
+    x.length # Expected error: length not defined for Integer
+  end
+end
+
+## Example connectives
+## success
+sig { params(x: T.any(String, Integer)).returns(Integer) }
+def connectives_success_f(x)
+  if !x.is_a?(Integer)
+    x.length
+  else
+    0
+  end
+end
+
+## success
+sig { params(x: T.untyped).returns(Integer) }
+def connectives_success_g(x)
+  if x.is_a?(String) || x.is_a?(Integer)
+    connectives_success_f(x)
+  else
+    0
+  end
+end
+
+## success
+sig { params(x: T.any(String, Integer, TrueClass, FalseClass)).returns(Integer) }
+def connectives_success_h(x)
+  if !x.is_a?(TrueClass) && !x.is_a?(FalseClass) && !x.is_a?(Integer)
+    x.length
+  else
+    0
+  end
+end
+
+## failure
+sig { params(x: T.any(String, Integer)).returns(Integer) }
+def connectives_failure_f(x)
+  if !x.is_a?(Integer)
+    x.is_nan # Expected error: is_nan not defined for String
+  else
+    0
+  end
+end
+
+## failure
+sig { params(x: T.untyped).returns(Integer) }
+def connectives_failure_g(x)
+  if x.is_a?(String) || x.is_a?(Integer)
+    x.length # Expected error: length not defined for String | Integer
+  else
+    0
+  end
+end
+
+## failure
+sig { params(x: T.any(String, Integer, TrueClass, FalseClass)).returns(Integer) }
+def connectives_failure_h(x)
+  if !x.is_a?(TrueClass) && !x.is_a?(FalseClass) && !x.is_a?(Integer)
+    x.is_nan # Expected error: is_nan not defined for String
+  else
+    0
+  end
+end
+
+## Example nesting_body
+## success
+sig { params(x: T.any(String, Integer, TrueClass, FalseClass)).returns(Integer) }
+def nesting_body_success_f(x)
+  if !x.is_a?(String)
+    if !x.is_a?(TrueClass) && !x.is_a?(FalseClass)
+      x + 1
+    else
+      0
+    end
+  else
+    0
+  end
+end
+
+## failure
+sig { params(x: T.any(String, Integer, TrueClass, FalseClass)).returns(Integer) }
+def nesting_body_failure_f(x)
+  if x.is_a?(String) || x.is_a?(Integer)
+    if x.is_a?(Integer)
+      x.length # Expected error: length not defined for Integer
+    else
+      0
+    end
+  else
+    0
+  end
+end
+
+## Example nesting_condition
+## success
+sig { params(x: T.untyped, y: T.untyped).returns(Integer) }
+def nesting_condition_success_f(x, y)
+  if x.is_a?(Integer) ? y.is_a?(String) : false
+    x + y.length
+  else
+    0
+  end
+end
+
+## failure
+sig { params(x: T.any(String, Integer), y: T.untyped).returns(Integer) }
+def nesting_condition_failure_f(x, y)
+  if x.is_a?(Integer) ? y.is_a?(String) : y.is_a?(String)
+    x.length # Expected error: length not defined for String | Integer
+  else
+    0
+  end
+end
+
+## Example predicate_2way
+## success
```

Comment:
Each section should have exactly 1 `success` comment and 1 `failure` comment. Try fixing & rerun the _benchmark file, does it work now?